zkevm-circuits/src/evm_circuit.rs

```diff
@@ -82,41 +78,64 @@ impl<F: Field> EvmCircuit<F> {
         )
     }
 
-    /// Assign block
-    pub fn assign_block(
-        &self,
-        layouter: &mut impl Layouter<F>,
-        block: &Block<F>,
-    ) -> Result<(), Error> {
-        self.execution.assign_block(layouter, block)
+    /// Load byte table
+    pub fn load_byte_table(&self, layouter: &mut impl Layouter<F>) -> Result<(), Error> {
+        layouter.assign_region(
+            || "byte table",
+            |mut region| {
+                for offset in 0..256 {
+                    region.assign_fixed(
+                        || "",
+                        self.byte_table[0],
+                        offset,
+                        || Ok(F::from(offset as u64)),
+                    )?;
+                }
+
+                Ok(())
+            },
+        )
     }
 
-    /// Assign exact steps in block without padding for unit test purpose
-    pub fn assign_block_exact(
+    /// Assign block
+    /// When exact is enabled, assign exact steps in block without padding for
+    /// unit test purpose
+    pub fn assign_block(
         &self,
         layouter: &mut impl Layouter<F>,
         block: &Block<F>,
+        exact: bool,
```

Comment:
That's probably be a good idea. I merged them into one to remove code duplication, but it does seem like it would be better to have two separate functions, with one only usable in tests, that call into this general one. The normal one always passing in `exact: false` and the test one always passing in `exact: true`. How does that sound?

Reply:
Yes, I mainly see two solutions:

1. You have a private internal fn which has the bool field and you expose two public fns which pass the `true` or `false` and mark one with `#[cfg(test)]` (More work IMO).

2. You have two functions with the exact same name and signature, and internally you define the logic taking into account the bool implications. You then mark one fn (the one with true IIUC) with `#[cfg(test)]` and you mark the other with: `#[cfg(not(test))]`

So that the compiler treats them as mutably exclusive. 

Whatever you prefer will be better than having this bool in the API :)

Reply:
I picked option 1 because I think it's probably useful to allow both variants to be used when testing (if they behave differently it could potentially hide some problems in the non-test version that may need specific testing).

Reply:
Hmm makes sense!!! LGTM then!!!